src/com/facebook/buck/apple/project_generator/ProjectGenerator.java

```diff
@@ -2685,7 +2695,14 @@ ProductType bundleToTargetProductType(
             return ProductType.APPLICATION;
         }
       } else if (binaryNode.getType().equals(AppleTestDescription.TYPE)) {
-        return ProductType.UNIT_TEST;
+        @SuppressWarnings("unchecked")
+        TargetNode<AppleTestDescription.Arg> testNode =
+              (TargetNode<AppleTestDescription.Arg>) binaryNode;
+        if (testNode.getConstructorArg().isUiTest()) {
```

Comment:
There is `TargetNode.castArg()` method that is usually used to cast the arg, and it returns `Optional.absent()` if cast is not possible. Should it help you here?